kratos/containers/geometry_container.h

```diff
@@ -197,11 +193,8 @@ class GeometryContainer
     /// Returns a reference geometry corresponding to the name
     TGeometryType& GetGeometry(std::string GeometryName)
     {
-        auto hash_index = TGeometryType::GenerateId(GeometryName);
-        auto i = mGeometries.find(hash_index);
-        KRATOS_ERROR_IF(i == mGeometries.end())
-            << " geometry index not found: " << GeometryName << ".";
-        return *i;
+        const auto& const_this = *this;
+        return const_cast<TGeometryType&>(const_this.GetGeometry(GeometryName));

```

Comment:
Please don't use `const_cast`

We usually copy-paste (yes you read correctly) the code, in most cases it is not much

There are some ways to avoid the code duplication, but they are not nice

Reply:
Now it's you against @pooyan-dadvand. I agree, but code duplication is not nice and this might appear in many functions. Some of those are a bit longer...

Reply:
well I too my "knowledge" from looking at @pooyan-dadvand s code
I also did some research not too long ago, as I said, there are not really nice solutions to do it without duplication

Reply:
C++ 17 will bring something. However, I don't think that this cast is for free either...

Reply:
I removed all const_cast to avoid troubles.